File: tools/get_cpu_info.py
"""CPU information tool."""

from tools import tool
from tools._env_helpers import fmt


@tool(
    name="get_cpu_info",
    description=(
        "Return CPU information: processor model, physical cores, logical cores."
    ),
    parameters={"type": "object", "properties": {}, "required": []},
)
def get_cpu_info() -> str:
    return fmt("cpu")

File: tools/get_cpu_info.ts
/** CPU information tool. */

import { registerInfoTool } from "./lib/env_helpers.js";

registerInfoTool({
  name: "get_cpu_info",
  description:
    "Return CPU information: processor model, physical/logical cores, max/current clock speed, L2/L3 cache, architecture, load percentage, virtualization status.",
  category: "cpu",
});
